mlflow/pyfunc/__init__.py

```diff
@@ -1138,7 +1138,15 @@ def _load_model_or_server(
     try:
         client.wait_server_ready(timeout=90, scoring_server_proc=scoring_server_proc)
     except Exception as e:
-        raise MlflowException("MLflow model server failed to launch.") from e
+        if scoring_server_proc.poll() is None:
+            # the scoring server is still running but client can't connect to it.
+            # kill the server.
+            scoring_server_proc.kill()
```

Comment:
```suggestion
            scoring_server_proc.kill()
            scoring_server_proc.wait()
```

Reply:
according to the logic here, we should kill it.

because `client.wait_server_ready(timeout=90, scoring_server_proc=scoring_server_proc)` raise error, it means the server is broken.

Reply:
@WeichenXu123 We should wait after kill

Reply:
oh sorry I misread it.

do we need wait after kill ? kill can be done in async way.